src/aiida/storage/sqlite_dos/backend.py

```diff
@@ -102,7 +102,7 @@ class Model(BaseModel):
         filepath: str = Field(
             title='Directory of the backend',
             description='Filepath of the directory in which to store data for this backend.',
-            default_factory=lambda: AIIDA_CONFIG_FOLDER / 'repository' / f'sqlite_dos_{uuid4().hex}',
+            default_factory=lambda: str(AIIDA_CONFIG_FOLDER / 'repository' / f'sqlite_dos_{uuid4().hex}'),
```

Comment:
Is this a functional change?

Reply:
This was actually a bug that went unnoticed. It is now tested for though by actually creating a profile with this backend in the unit tests